src/vocab_tree/vocab_tree.cpp

```diff
@@ -42,36 +42,38 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "vocab_tree.h"
 #include "clustering.h"
 
+#include <Eigen/Dense>
+
 using std::cout;
 using std::endl;
 
-float l2sq(DTYPE *a, DTYPE *b, int dim)
```

Comment:
don't know why but this change on l2sq is much slower on my mac compare to the original implementation